### PR TITLE
Publicize: Update facebook preview

### PR DESCRIPTION
**Pull request by monsieur-z**
Fixes 1203820933396971-as-1204281808721008

## Proposed changes:

This PR updates the props passed to the `FacebookPreview` component to match the next version of the component.

_Note: the new version of the package containing the `FacebookPreview` component hasn't been published yet. Do not merge this PR yet._

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

[Project thread](pdrWKz-Ka-p2)

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* Checkout this PR branch
* Checkout the latest `trunk` in calypso
* In `wp-calypso/packages/social-previews/` directory, run `pnpm link --global` to expose the package to global `node_modules` for `pnpm`
* In `jetpack/projects/js-packages/publicize-components/` directory, run `pnpm link --global @automattic/social-previews`
* In `wp-calypso` root, run `yarn` and then `yarn workspace @automattic/social-previews run build`
* In Jetpack, run `jetpack build plugins/jetpack --no-pnpm-install`
* Start writing a new post with JT site via Jetpack
* Click Jetpack icon on the top of the sidebar
* Under "Social Previews", click on "Preview"
* Check that you see the new Facebook preview and that there's no error in the console
* Once you are done, run `pnpm unlink` in Jetpack

### CODE REVIEW

**Comment by manzoorwanijk** [comment]
The base branch needs to be `feature/social-previews-v2`

File: projects/js-packages/publicize-components/src/components/social-previews/constants.js [hunk 1]
```diff
@@ -12,7 +12,37 @@ export const AVAILABLE_SERVICES = [
...
+				<FacebookPreview
+					{ ...props }
+					type="article"
+					user={ user }
+					customText={ message || excerpt || content || title }
+					customImage={ sourceUrl }
+					imageMode={ imageMode }
+				/>
+			);
+		},
```
Comment (manzoorwanijk) [inline projects/js-packages/publicize-components/src/components/social-previews/constants.js]: May be we can extract this to a separate component and keep the constants file cleaner?

File: projects/js-packages/publicize-components/src/components/social-previews/modal.js [hunk 2]
```diff
@@ -58,6 +70,12 @@ const SocialPreviewsModal = function SocialPreviewsModal( {
 							image={ image }
 							isTweetStorm={ isTweetStorm }
 							tweets={ tweets }
+							connection={ connections?.find( conn => conn.service_name === tab.name ) }
+							excerpt={ excerpt }
+							content={ content }
+							message={ message }
+							mediaDetails={ mediaDetails }
+							attachedMediaDetails={ attachedMediaDetails }
```
Comment (manzoorwanijk) [inline projects/js-packages/publicize-components/src/components/social-previews/modal.js]: If we create a separate component here for facebook and use it in constants, we can move these props to that component. From here, they get passed to every preview component.

**Comment by monsieur-z** [comment]
I followed your suggestions @manzoorwanijk.

**Review by manzoorwanijk — APPROVED** [review]
This looks good now. Failing build is due to calypso package not published yet.

**Comment by monsieur-z** [comment]
> Failing build is due to calypso package not published yet.

Is it still okay to merge it with `feature/social-previews-2`, @manzoorwanijk?

**Comment by manzoorwanijk** [comment]
> Is it still okay to merge it with `feature/social-previews-2`, @manzoorwanijk?

Yes, it is

**Pull request merged**